### PR TITLE
Add Amazon S3 ACL Support

### PR DESCRIPTION
**Pull request by shichanglin5**
# What problem are we solving?

[Add Amazon S3 ACL Support](https://github.com/seaweedfs/seaweedfs/pull/3719)

# How are we solving the problem?

# How is the PR tested?

Test through http interface using s3-sdk

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/pb/iam.proto [hunk 1]
```diff
@@ -20,8 +20,9 @@ message S3ApiConfiguration {
 
 message Identity {
     string name = 1;
-    repeated Credential credentials = 2;
-    repeated string actions = 3;
+    string account = 2;
+    repeated Credential credentials = 3;
```
Comment (chrislusf) [inline weed/pb/iam.proto:24]: Do not change field index.

**Review by chrislusf — COMMENTED** [review]
The PR is too big to review. Please split this into multiple PRs. For each change, you can have one PR to refactor the code and another PR to make the logic change.

**Comment by shichanglin5** [comment]
Okay

**Comment by shichanglin5** [comment]
> The PR is too big to review. Please split this into multiple PRs. For each change, you can have one PR to refactor the code and another PR to make the logic change.

This pr split is a bit troublesome, because there are a lot of couplings, so I split it into two parts, the first is this PR #3731  (ACL related structure introduction)

**Comment by shichanglin5** [comment]
> > The PR is too big to review. Please split this into multiple PRs. For each change, you can have one PR to refactor the code and another PR to make the logic change.
> 
> This pr split is a bit troublesome, because there are a lot of couplings, so I split it into two parts, the first is this PR #3731 (ACL related structure introduction)

When this PR is no problem, I will submit another PR on this basis (ACL logic implementation and related interface modification)

**Pull request closed without merging**